radixdlt-core/radixdlt/src/test/java/com/radixdlt/client/store/berkeley/BerkeleyClientApiStoreTest.java

```diff
@@ -0,0 +1,102 @@
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+package com.radixdlt.client.store.berkeley;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.collect.ImmutableMap;
+import com.radixdlt.DefaultSerialization;
+import com.radixdlt.atom.SpunParticle;
+import com.radixdlt.atommodel.tokens.StakedTokensParticle;
+import com.radixdlt.atommodel.tokens.TokenPermission;
+import com.radixdlt.atommodel.tokens.TransferrableTokensParticle;
+import com.radixdlt.identifiers.RRI;
+import com.radixdlt.identifiers.RadixAddress;
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.store.DatabaseEnvironment;
+import com.radixdlt.store.LedgerEntryStore;
+import com.radixdlt.utils.UInt256;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+
+import static com.radixdlt.atommodel.tokens.MutableSupplyTokenDefinitionParticle.TokenTransition;
+
+public class BerkeleyClientApiStoreTest {
+	private static final RadixAddress OWNER = RadixAddress.from("JEbhKQzBn4qJzWJFBbaPioA2GTeaQhuUjYWkanTE6N8VvvPpvM8");
+	private static final RadixAddress DELEGATE = RadixAddress.from("JH1P8f3znbyrDj8F4RWpix7hRkgxqHjdW2fNnKpR3v6ufXnknor");
+	private static final RadixAddress TOKEN_ADDRESS = RadixAddress.from("23B6fH3FekJeP6e5guhZAk6n9z4fmTo5Tngo3a11Wg5R8gsWTV2x");
+	private static final RRI TOKEN = RRI.of(TOKEN_ADDRESS, "COOKIE");
+	private static final UInt256 GRANULARITY = UInt256.ONE;
+	private static final ImmutableMap<TokenTransition, TokenPermission> TOKEN_PERMISSIONS = ImmutableMap.of();
+
+	private final Serialization serialization = DefaultSerialization.getInstance();
+	private final LedgerEntryStore ledgerStore = mock(LedgerEntryStore.class);
+
+	private DatabaseEnvironment environment;
+
+	@Rule
+	public TemporaryFolder folder = new TemporaryFolder();
+
+	@Before
+	public void setUp() {
+		environment = new DatabaseEnvironment(folder.getRoot().getAbsolutePath(), 0);
+	}
+
+	@Test
+	public void tokenBalancesAreReturned() {
+		var particles = List.of(SpunParticle.up(stake(UInt256.TWO)), SpunParticle.up(transfer(UInt256.NINE)));
+		var clientApiStore = prepareApiStore(particles);
+
+		clientApiStore.getTokenBalances(OWNER)
+			.onSuccess(list -> {
+				assertEquals(2, list.size());
```

Comment:
I **think** this test is double incorrect, but I might have misunderstood something?:

# 1. Balances should be reduced/aggregated/merged *per token* (RRI)
Why does this test return a list of length 2? Both your two helper methods `stake(UInt256 amount)` and `transfer(UInt256 amount)` respectively use the **same** `RRI`, so the list should really just have contained one entry

We should NOT return these two balance entries

```
[
     { amount: 3, rri: "ALICE_TOKEN" }, 
     { amount: 5, rri: "ALICE_TOKEN" }, 
]
``` 

for these particles (pseudocode):

```
[
	SpunParticle {
		spin: UP,
		particle: {
			type: TransferrableTokensParticle,
			rri: ALICE_TOKEN
			amount: 3
		}
	},
	SpunParticle {
		spin: UP,
		particle: {
			type: TransferrableTokensParticle,
			rri: ALICE_TOKEN
			amount: 5
		}
	},
]
```

Instead, we should return the single item (pseudocode):

```
[
     { amount: 8, rri: "ALICE_TOKEN" }, 
]
```

Being the calculated total balance of the `"ALICE_TOKEN"`.

# 2. Staked amount should be _subtracted_ 
All staked tokens should considered as "spent", or rather, considered as _not  **immediately** spendable_

So this test should be update to this:

```java
clientApiStore.getTokenBalances(OWNER)
			.onSuccess(list -> {
				assertEquals(1, list.size());
				assertEquals(UInt256.NINE, list.get(0).getAmount());
				assertEquals(TOKEN, list.get(0).getRri());
			})
```

Just a **single entry per token (RRI)** and having the amount `NINE`, since that is what was transferred to the owner,, the `TWO` is staked, so not spendable.

Reply:
Fixed